### DISCUSSION TITLE
How can I react to GRPC exporter client errors

### DISCUSSION THREAD

**Discussion by ivanaguilario** [open-telemetry/opentelemetry-go#3770] [Q&A] [answered]
I added the OTLP GRPC exporter to my service. I'd like to be able to react if the grpc client used by the exporter throws any error, as I don't want my service to run at all if I'm not going to receive spans in my collector.

One such error would be for example a timeout when I don't specify a port for the GRPC endpoint.

Any ideas?

This is my code to initialize the trace provider:

```go
func initOTELTracerProvider(ctx context.Context, resource *resource.Resource) (func(context.Context) error, error) {
	connCtx, cancel := context.WithTimeout(ctx, time.Second)
	defer cancel()

	otlpExporter, err := otlptracegrpc.New(connCtx, otlptracegrpc.WithTimeout(time.Second*5))

	if err != nil {
		return nil, err
	}

	otlpProcessor := sdktrace.NewBatchSpanProcessor(otlpExporter)

	tracerProvider := sdktrace.NewTracerProvider(
		sdktrace.WithSampler(sdktrace.AlwaysSample()),
		sdktrace.WithResource(resource),
		sdktrace.WithSpanProcessor(otlpProcessor),
	)

	otel.SetTracerProvider(tracerProvider)

	otel.SetTextMapPropagator(propagation.TraceContext{})

	ctx, span := otelTracer.Start(ctx, "initOTELTracerProvider")
	defer span.End()

	return tracerProvider.Shutdown, nil
}
```

**Comment by dmathieu** [comment, marked as answer]
All errors should be sent to the [ErrorHandler](https://pkg.go.dev/go.opentelemetry.io/otel#SetErrorHandler) which you can be set as you see fit.

**Reply by ivanaguilario** [reply]
Thank you very much, this was exactly what I was looking for!
